src/main/java/slimeknights/tconstruct/library/modifiers/Modifier.java

```diff
@@ -324,6 +330,41 @@ public void onInventoryTick(IModifierToolStack tool, int level, World world, Liv
   }
 
 
+  /**
+    * Called when this item is used when targeting a Block
+   * @param tool 
+   * @param level 
+   * @param stack 
+   * @param context        Full item use context
+   * @return  Action result
+   */
+  public ActionResultType onItemUse(ToolStack tool, int level, ItemStack stack, ItemUseContext context) {
```

Comment:
Per below comment, any reason you are using `ToolStack` directly? Can we switch it to the interface and switch callers to support the interface as well? Likewise, do we need `ItemStack` here, or can they just fetch it from the context?

Reply:
ToolHarvestEvent and ToolHarvestLogic.getAOEBlocks.

Reply:
Change those methods to take an `IModifierToolStack`, let me know if anything there is not possible with the interface.

For the stack, if we need it we can always fetch from the context, just goes back to discouraging using it directly unless needed